File: Sorting/Merge/Merge.c
#include <stdio.h>
void merge(int arr[],int left,int mid,int right){
    int len1=mid-left+1,len2=right-mid;
    int left_arr[len1],right_arr[len2],index_arr1=0,index_arr2=0,index=left;
    for(int i=left;i<=mid;i++)
    left_arr[i-left]=arr[i];
    for(int i=mid+1;i<=right;i++)
    right_arr[i-mid-1]=arr[i];
    while(index_arr1<len1 && index_arr2<len2){
        if(left_arr[index_arr1]<right_arr[index_arr2])
        arr[index++]=left_arr[index_arr1++];
        else
        arr[index++]=right_arr[index_arr2++];
    }
    while(index_arr1<len1)
    arr[index++]=left_arr[index_arr1++];
    while(index_arr2<len2)
    arr[index++]=right_arr[index_arr2++];
}
void mergesort(int arr[],int left,int right){
   if(left<right){
        int mid=left+(right-left)/2;
        mergesort(arr,left,mid);
        mergesort(arr,mid+1,right);
        merge(arr,left,mid,right);
    }
}
int main(void) {
    int arr[]={4,2,6,1,5,3};
    mergesort(arr,0,5);
    for(int i=0;i<6;i++)
    printf("%d ",arr[i]);
	return 0;
}

File: Sorting/Merge/Merge.py
class sort:
     def __init__(self):
          pass
     def merge(self,l,left,mid,right):
          len1,len2=mid-left+1,right-mid
          left_arr=l[left:mid+1]
          right_arr=l[mid+1:right+1]
          index_arr1,index_arr2,index=0,0,left
          while index_arr1<len1 and index_arr2<len2:
               if left_arr[index_arr1]<right_arr[index_arr2]:
                    l[index]=left_arr[index_arr1]
                    index_arr1+=1
               else:
                    l[index]=right_arr[index_arr2]
                    index_arr2+=1
               index+=1
          while index_arr1<len1:
               l[index]=left_arr[index_arr1]
               index_arr1+=1
               index+=1
          while index_arr2<len2:
               l[index]=right_arr[index_arr2]
               index_arr2+=1
               index+=1
          return l
     def merge_sort(self,l,left,right):
          if left<right:
               mid=left+((right-left)//2)
               l=self.merge_sort(l,left,mid)
               l=self.merge_sort(l,mid+1,right)
               l=self.merge(l,left,mid,right)
          return l
l=[ 7, 3, 5, 6, 4, 2 ]
m_sort=sort()
l=m_sort.merge_sort(l,0,len(l)-1)
print(*l)